torch/csrc/deploy/interpreter/interpreter_impl.cpp

```diff
@@ -249,10 +249,8 @@ struct __attribute__((visibility("hidden"))) ConcreteInterpreterImpl
 #define APPEND_INIT(name) PyImport_AppendInittab(#name, PyInit_##name);
     FOREACH_LIBRARY(APPEND_INIT)
 #undef APPEND_INIT
-    PyImport_AppendInittab("torch._C", initModule);
-    if (PyInit_tensorrt) {
-      PyImport_AppendInittab("tensorrt.tensorrt", PyInit_tensorrt);
-    }
+
+    BuiltinRegistry::appendCPythonInittab();
```

Comment:
Here, it seems we are doing three things:
- Appending to the builtin module list (inittab)
- Sanity check the number of Python modules is what we expect
- Setting `Py_FrozenModules` to our new set of frozen modules.

I wonder if it makes more sense to encapsulate this logic in `BuiltinRegistry`. Overall, the flow looks like:
1. Builtin preinitialization (the 3 steps above)
2. Initialize the python inteprreter
3. Builtin postinitialization (e.g. doing the sys.meta_path trick)

So maybe we should have a `BuiltinRegistry::runPreInitialization` that does step 1 (and checks that `Py_IsInitialized()` is false, etc.) and a `BuiltinRegistry::runPostInitialization`, that does step 3.

That way we can make the interaction between interpreter initialization and `BuiltinRegistry` simple. What do you think?